### ISSUE TITLE
[Feature Request]align rspack plugin with webpack plugin

### ISSUE THREAD

**Issue by maroon1** [nrwl/nx-labs#377]
There are many rspack config that is compatible with webpack, it's great if rspack plugin align with webpack plugin. 

I find that rspack plugin use `cssModule` option to determine how css module work, but webpack plugin define two loader rules to sperately handle global css and css module. It result hard to migrate to rspack from webpack

in rspack plugin

https://github.com/nrwl/nx-labs/blob/271625ed814405ff852906f64642e7050f5632be/packages/rspack/src/utils/with-web.ts#L47-L98

in webpack plugin

https://github.com/nrwl/nx/blob/a1d8645ac57864ba1a7347b9f288f3052aba2263/packages/webpack/src/plugins/nx-webpack-plugin/lib/apply-web-config.ts#L128-L243

**Comment by phal0r** [comment]
I agree. This logic is flawed and i could be really easy, because rspack handles css out of the box and it is not even necessary to specify it (https://www.rspack.dev/guide/migrate-from-webpack.html#removing-css-loader-style-loader-and-mini-css-extract-plugin).

Even if it is specified, setting to undefined didn't work in my tests, but you can just set the type to 'css/auto' and rspack will handle style files without ".module" in the name as normal css and files with ".module" in the name as css modules (https://www.rspack.dev/guide/language-support.html#css).

The situation gets worse, when you use withReact instead of withWeb, because withReact loads withWeb internally and forces cssModules, no matter what you pass in as options -> https://github.com/nrwl/nx-labs/blob/main/packages/rspack/src/utils/with-react.ts#L14

All of this should be fixed and configurable, so we don't need to crawl deep into the sourcecode to understand, why the bundling behaves weird.
